x/streamer/types/distr_info.go

```diff
@@ -36,29 +35,19 @@ func (r DistrRecord) ValidateBasic() error {
 	return nil
 }
 
-var hundred = math.NewInt(100)
-
-// DistrInfoFromDistribution converts sponsorship distribution to the DistrInfo type and performs a
-// basic validation for DistrInfo.Records. Returning an empty DistrInfo (with zero DistrInfo.TotalWeight)
-// is a valid scenario.
+// DistrInfoFromDistribution converts sponsorship distribution to the DistrInfo type.
+// Returning an empty DistrInfo (with zero DistrInfo.TotalWeight) is a valid scenario.
 func DistrInfoFromDistribution(d sponsorshiptypes.Distribution) *DistrInfo {
-	totalWeight := math.ZeroInt()
 	records := make([]DistrRecord, 0, len(d.Gauges))
 	for _, g := range d.Gauges {
-		// d.VotingPower is always > 0 to the Distribution type contract
-		weight := g.Power.Mul(hundred).Quo(d.VotingPower)
-
-		record := DistrRecord{
+		records = append(records, DistrRecord{
 			GaugeId: g.GaugeId,
-			Weight:  weight,
-		}
-
-		totalWeight = totalWeight.Add(weight)
-		records = append(records, record)
+			Weight:  g.Power,
+		})
 	}
 
 	return &DistrInfo{
-		TotalWeight: totalWeight,
+		TotalWeight: d.VotingPower,
 		Records:     records,
 	}
 }
```

Comment:
This should really be a method!
I asked for this before you ignored it!

Generally the development world accepted OOP as a good thing in most cases. It makes code easier to read, requires less verbose naming, allows better autocomplete, and gives better understanding of package functionality when viewing structural layout of a packge

Reply:
This should be a method .Info()